src/cascadia/TerminalSettingsModel/MatchProfilesEntry.cpp

```diff
@@ -43,22 +43,31 @@ winrt::com_ptr<NewTabMenuEntry> MatchProfilesEntry::FromJson(const Json::Value&
 
 bool MatchProfilesEntry::MatchesProfile(const Model::Profile& profile)
 {
-    auto isMatching = false;
+    // We use an optional here instead of a simple bool directly, since there is no

```

Comment:
I kinda wonder if we want the opposite behavior! Matching with _no conditions_ means everything passes... that would let us simplify the logic as in my farther-down comment

Reply:
@zadjii-msft Does that make sense to you, or did I countermand you?

Reply:
Huh. I've been thinking about it all night, and honestly both seem alright, so I think we should just pick one and go with it. 

Having no conditions match _everything_ kinda makes sense - you start with everything, and each property filters that list down more.

Having it match _nothing_ also makes sense - there's no properties, so that doesn't really make sense as an entry, and then it just gets ignored. 

(also, I've got the plague now too so I might need @dhowett to finish driving this one down)

Reply:
If we do decide for default everything, then naming it `filterProfiles` makes more sense. So then, `matchProfiles` to me is default nothing.